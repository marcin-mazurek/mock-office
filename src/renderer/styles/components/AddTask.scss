.add-task {
  height: 100%;
  overflow-y: scroll;
}

.add-task__back-button {
  color: inherit;
}

.add-task__main {
  padding: 15px;
}
