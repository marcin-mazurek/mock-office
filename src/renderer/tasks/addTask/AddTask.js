import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router';
import { getSelectedServerDetails } from '../../servers/selectors';
import Http from './HttpForm';
import Ws from './WsForm';

export const AddTask = ({ serverType }) => (
  <div className="add-task">
    <Link to="/">&#8666; Server details</Link>
    {
      serverType === 'http'
        ? <Http />
        : <Ws />
    }
  </div>
);

AddTask.propTypes = {
  serverType: React.PropTypes.string.isRequired
};

const mapStateToProps = state => ({
  serverType: getSelectedServerDetails(state).type
});

export default connect(mapStateToProps)(AddTask);
