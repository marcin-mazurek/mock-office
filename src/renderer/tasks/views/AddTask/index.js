import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router';
import { getSelectedServerDetails } from '../../../servers/selectors';
import Http from './httpForm';
import Ws from './wsForm';

const AddTaskForm = ({ serverType }) => (
  <div className="add-task">
    <Link to="/server">&#8666; Server details</Link>
    {
      serverType === 'http'
        ? <Http />
        : <Ws />
    }
  </div>
);

AddTaskForm.propTypes = {
  serverType: React.PropTypes.string.isRequired
};

const mapStateToProps = state => ({
  serverType: getSelectedServerDetails(state).type
});

export default connect(mapStateToProps)(AddTaskForm);
