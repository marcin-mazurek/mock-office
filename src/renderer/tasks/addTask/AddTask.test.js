import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { AddTask } from './AddTask';

describe('AddTask', () => {
  test('default snapshot', () => {
    const props = {
      serverType: 'some type'
    };
    const wrapper = shallow(<AddTask {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
