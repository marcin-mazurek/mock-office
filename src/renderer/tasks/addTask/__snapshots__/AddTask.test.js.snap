exports[`AddTask default snapshot 1`] = `
<div
  className="add-task">
  <Link
    onlyActiveOnIndex={false}
    style={Object {}}
    to="/">
    â‡š Server details
  </Link>
  <Connect(WsForm) />
</div>
`;
